### PR TITLE
[FLINK-23172][docs] Fix broken links to Task Failure Recovery page on Configuration page

### PR DESCRIPTION
**Pull request by Thesharing**
## What is the purpose of the change

*In #16624 we attempt to fix broken links to Task Failure Recovery page on Configuration page. The broken links on the Configuration page are fixed. But the links on Task Failure Recovery page itself are not. We didn't notice it before. Thus, we need to propose a better way to fix the broken links. In Hugo we can use the variable `{{.Site.BaseURL}}` to make sure the links are always right no matter where they are. This variable makes sure the links refer to absolute path instead of relative path.*

Ref: https://gohugo.io/variables/site/#site-variables-list

## Brief change log

- *Fix broken links in `RestartStrategyOptions.java` and `JobManagerOptions.java`*
- *Generate the docs based on the modification above*

## Verifying this change

*Run the local server with Hugo and check if the links head to the correct pages.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by Thesharing** [comment]
Hi, @sjwiesman. Since I'm not that familiar with Hugo. Could you please help me review this pull request? Do I just use the variable  `{{.Site.BaseURL}}` correctly?

File: flink-core/src/main/java/org/apache/flink/configuration/RestartStrategyOptions.java
Comment (zhuzhurk) [inline flink-core/src/main/java/org/apache/flink/configuration/RestartStrategyOptions.java:80]: maybe `{{.Site.BaseURL}}/docs` to be safer?

Comment (Thesharing) [reply flink-core/src/main/java/org/apache/flink/configuration/RestartStrategyOptions.java:80]: > maybe `{{.Site.BaseURL}}/docs` to be safer?

Yes, it will make sure the URL won't be broken if `{{.Site.BaseURL}}` is not ended with `/`. But for now it will make the URL as `https://ci.apache.org/projects/flink/flink-docs-master//docs/ops/state/task_failure_recovery/`, would it be okay?

Comment (zhuzhurk) [reply flink-core/src/main/java/org/apache/flink/configuration/RestartStrategyOptions.java:80]: I took another look and can find {{.Site.BaseURL}}  sometimes used with trailing '/' and sometimes not. I think at the moment we can keep the change as is.

**Review by zhuzhurk — APPROVED** [review]
LGTM.

**Comment by Thesharing** [comment]
I've already rebased the latest master branch and verify the changes manually again. Waiting for CI to give a green light...

File: flink-core/src/main/java/org/apache/flink/configuration/RestartStrategyOptions.java
Comment (zhuzhurk) [reply flink-core/src/main/java/org/apache/flink/configuration/RestartStrategyOptions.java:80]: After merging this PR, the link becomes "https://ci.apache.org/projects/flink/flink-docs-masterdocs/ops/state/task_failure_recovery#fixed-delay-restart-strategy" and still breaks.
I think "flink-docs-masterdocs" is not expected and a `/` is still needed.

**Pull request closed without merging**
